src/mem/ruby/protocol/chi/CHI-cache-funcs.sm

```diff
@@ -892,10 +892,10 @@ void copyCacheAndDir(CacheEntry cache_entry, DirEntry dir_entry,
   if (is_valid(cache_entry) &&
       ((initialState == State:UD) || (initialState == State:SD) ||
         (initialState == State:UC) || (initialState == State:SC) ||
-        (initialState == State:UD_RSC) || (initialState == State:SD_RSC) ||
-        (initialState == State:UC_RSC) || (initialState == State:SC_RSC) ||
-        (initialState == State:UD_RSD) || (initialState == State:SD_RSD) ||
-        (initialState == State:UD_T))) {
+        (initialState == State:UD_RU) || (initialState == State:UD_RSC) ||
```

Comment:
We don't set dataValid=true if any upstream cache has the data in unique state to prevent the local copy from being used (because it may have been modified upstream). This changes allows dataValid=true when the state UD_RU. I think this can generate inconsistency when handling other transactions.